src/System.Net.Primitives/src/System/Net/CookieCollection.cs

```diff
@@ -0,0 +1,297 @@
+                {
+                    if (string.Compare(c.Name, name, StringComparison.OrdinalIgnoreCase) == 0)
+                    {
+                        return c;
+                    }
+                }
+                return null;
+            }
+        }
+
+        public void Add(Cookie cookie)
+        {
+            if (cookie == null)
+            {
+                throw new ArgumentNullException("cookie");
+            }
+            ++_version;
+            int idx = IndexOf(cookie);
+            if (idx == -1)
+            {
+                _list.Add(cookie);
+            }
+            else
+            {
+                _list[idx] = cookie;
+            }
+        }
+
+        public void Add(CookieCollection cookies)
+        {
+            if (cookies == null)
+            {
+                throw new ArgumentNullException("cookies");
+            }
+            foreach (Cookie cookie in cookies)
+            {
+                Add(cookie);
+            }
+        }
+
+        public int Count
+        {
+            get
+            {
+                return _list.Count;
+            }
+        }
+
+        public bool IsSynchronized
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public object SyncRoot
+        {
+            get
+            {
+                return this;
+            }
+        }
+
+        public void CopyTo(Array array, int index)
+        {
+            _list.CopyTo(array, index);
+        }
+
+        public void CopyTo(Cookie[] array, int index)
+        {
+            _list.CopyTo(array, index);
+        }
+
+        internal DateTime TimeStamp(Stamp how)
+        {
+            switch (how)
+            {
+                case Stamp.Set:
+                    _timeStamp = DateTime.Now;
+                    break;
+                case Stamp.SetToMaxUsed:
+                    _timeStamp = DateTime.MaxValue;
+                    break;
+                case Stamp.SetToUnused:
+                    _timeStamp = DateTime.MinValue;
+                    break;
+                case Stamp.Check:
+                default:
+                    break;
+            }
+            return _timeStamp;
+        }
+
+
+        // This is for internal cookie container usage.
+        // For others not that _hasOtherVersions gets changed ONLY in InternalAdd
+        internal bool IsOtherVersionSeen
+        {
+            get
+            {
+                return _hasOtherVersions;
+            }
+        }
+
+        // If isStrict == false, assumes that incoming cookie is unique.
+        // If isStrict == true, replace the cookie if found same with newest Variant.
+        // Returns 1 if added, 0 if replaced or rejected.
+        internal int InternalAdd(Cookie cookie, bool isStrict)
+        {
+            int ret = 1;
+            if (isStrict)
+            {
+                IComparer comp = Cookie.GetComparer();
```

Comment:
Maybe. I'm going to leave this as-is for now: its equality rules differ from those of `Cookie` itself.